Изменено: создание трех диалогов с возможностью отмены и возврата на …

Maxim-ka opened the pull request:
…предыдущий диалог

Добавлен DialogFunctionPresenter для работы с диалогами

Maxim-ka committed:
Изменено: создание трех диалогов с возможностью отмены и возврата на предыдущий диалог

Добавлен DialogFunctionPresenter для работы с диалогами

AlinaRakhimova commented on app/src/main/java/ru/geekbrains/psy_journal/presenter/DialogFunctionPresenter.java:
```diff
@@ -0,0 +1,102 @@
+	}
+
+	private void getOTF(){
+		Disposable disposable = settable.getOTF()
```
А зачем здесь обращаться к settable? Нельзя сразу из RoomHelper-а эти данные брать?

KunakbaevAV replied:
Этот и следующий подобные методы я бы поменял 
settable.getOTF() 
на
roomHelper.getOTFList()

KunakbaevAV commented on app/src/main/java/ru/geekbrains/psy_journal/presenter/Settable.java:
```diff
@@ -1,8 +1,16 @@
-	void setUpdated(Updated updated);
+	Single<List<OTF>> getOTF();
+	Single<List<TF>> getTF(int idOTF);
+	Single<List<TD>> getTD(int idTF);
```
Считаю эти три метода в интерфейсе лишними, так как они есть в RoomHelper, который нужно просто инжектить в DialogFunctionPresenter и пользоваться этими методами там напрямую, а не через AddWorkPresenter.

KunakbaevAV commented on app/src/main/java/ru/geekbrains/psy_journal/presenter/DialogFunctionPresenter.java:
```diff
@@ -0,0 +1,102 @@
+	private final List<Functional> list = new ArrayList<>();
+	private Settable settable;
+	private String title;
+
```
Если инжектировать RoomHelper, почему-то, нужно добавлять инициализацию не в конструкторе, а статическую (тестировал у себя):
	@Inject
	RoomHelper roomHelper;

	{
		App.getAppComponent().inject(this);
	}

KunakbaevAV reviewed (approved):
По коду у меня есть только одно предложение. Описал в комментариях

Maxim-ka committed:
Внесены изменения:

добавлен roomHelper в презентер DialogFunctionPresenter
убаны методы выбора загрузки во фрагмент FunctionDialog в свзяи с примением
строковых ресурсов

Maxim-ka committed:
Добавлено: пункт "Иная деятельность" в список TF в БД

Реализовать условие в диалоговом окне
Отображение "Иная деятельность" в поле
статический конструктор AddWorkFragment для обьекта Journal

KunakbaevAV committed:
Проверка поля "Декларируемый запрос"
1. Сделал проверку на заполнение поля "Декларируемый запрос" в AddWorkFragment
2. Добавил библиотеку с анимациями

KunakbaevAV committed:
Merge branch 'MaximReschikov' of https://github.com/KunakbaevAV/psy_journal into ArtemKunakbaev

# Conflicts:
#	app/src/main/java/ru/geekbrains/psy_journal/view/MainActivity.java
#	app/src/main/java/ru/geekbrains/psy_journal/view/fragment/AddWorkFragment.java
#	app/src/main/res/values/strings.xml

Maxim-ka committed:
Добавлено: очистка backStack при добавлении новой единицы работы
